Library/Homebrew/livecheck/strategy/hackage.rb

```diff
@@ -18,7 +18,7 @@ module Strategy
       # @api public
       class Hackage
         # The `Regexp` used to determine if the strategy applies to the URL.
-        URL_MATCH_REGEX = /(?:downloads|hackage)\.haskell\.org/i.freeze
+        URL_MATCH_REGEX = %r{^https?://(?:downloads|hackage)\.haskell\.org(?:/[^/]+){3}}i.freeze
```

Comment:
Ensures that the path has exactly 3 parts because 3rd one is the filename used for package name extraction.
```rb
# Hackage URLs take one of the following formats:
#
# * `https://hackage.haskell.org/package/example-1.2.3/example-1.2.3.tar.gz`
# * `https://downloads.haskell.org/~ghc/8.10.1/ghc-8.10.1-src.tar.xz`
```